scipy/stats/stats.py

```diff
@@ -6147,8 +6147,19 @@ def power_divergence(f_obs, f_exp=None, ddof=0, axis=0, lambda_=None):
     # the test statistic.  We use some specialized code for a few special
     # cases of lambda_.
     if lambda_ == 1:
+        f_obs = f_obs.astype(np.float64)
```

Comment:
It's surprising all the things that can go wrong here. I ran into:

- overflow when data type is `uint8`
- warnings for divide by zero when `f_exp` is empty, NaN, etc.
- trouble summing when array is 0d
- incorrect sum in denominator when only f_obs is masked (for normalization)

so that's why this isn't a one line check. Ended up being relatively simple, all things considered.